src/ui/api/utils.go

```diff
@@ -316,7 +307,7 @@ func diffRepos(reposInRegistry []string, reposInDB []string,
 			if err != nil {
 				return needsAdd, needsDel, err
 			}
-			client, err := NewRepositoryClient(endpoint, true,
+			client, err := uiutils.NewRepositoryClientForUI(endpoint, true,
 				"admin", repoInR, "repository", repoInR, "pull")
```

Comment:
Should be uiutils.NewRepositoryClientForUI(endpoint, true,"admin", repoInR, "pull")